docs/streams/developer-guide/processor-api.html

```diff
@@ -396,12 +400,68 @@ <h2>
                 </ul>
                 </p>
             </div>
+            <div class="section" id="versioned-state-stores">
+                <span id="streams-developer-guide-state-store-versioned"></span><h3><a class="toc-backref" href="#id12">Versioned Key-Value State Stores</a><a class="headerlink" href="#versioned-state-stores" title="Permalink to this headline"></a></h3>
+                <p>Versioned key-value state stores are available since Kafka Streams 3.5.
+                    Rather than storing a single record version (value and timestamp) per key,
+                    versioned state stores may store multiple record versions per key. This
+                    allows versioned state stores to support timestamped retrieval operations
+                    to return the latest record (per key) as of a specified timestamp, which
+                    enables use cases such as temporal stream-table joins.</p>
+                <p>You can create a persistent, versioned state store by passing a
+                    <a class="reference external" href="/{{version}}/javadoc/org/apache/kafka/streams/state/Stores.html#persistentVersionedKeyValueStore(java.lang.String,java.time.Duration)">VersionedBytesStoreSupplier</a>
+                    to the
+                    <a class="reference external" href="/{{version}}/javadoc/org/apache/kafka/streams/state/Stores.html#versionedKeyValueStoreBuilder(java.lang.String,java.time.Duration)">versionedKeyValueStoreBuilder</a>,
+                    or by implementing your own
+                    <a class="reference external" href="/{{version}}/javadoc/org/apache/kafka/streams/state/VersionedKeyValueStore.html">VersionedKeyValueStore</a>.</p>
+                <p>Each versioned store has an associated, fixed-duration <em>history retention</em>
+                    parameter which specifies long old record versions should be kept for.
+                    In particular, a versioned store guarantees to return accurate results for
+                    timestamped retrieval operations where the timestamp being queried is within
+                    history retention of the current observed stream time.</p>
+                <p>History retention also doubles as its <em>grace period</em>, which determines
+                    how far back in time out-of-order writes to the store will be accepted. A
+                    versioned store will not accept writes (inserts, updates, or deletions) if
+                    the timestamp associated with the write is older than the current observed
+                    stream time by more than the grace period. Stream time in this context is
```

Comment:
`stream time` -> `stream-time` (multiple times)

Reply:
Hm, this looks to be inconsistent throughout the existing codebase since searching for both "stream time" and "stream-time" turn up a bunch of results across docs and also javadocs/code comments. Do we have an agreed upon standard for the repo? Personally I prefer "stream time" but unification is probably more important :) If there's already consensus I can make the updates in this PR. Otherwise, might be best left for a follow-up.